pkg/diagnostics/context.go

```diff
@@ -85,6 +87,50 @@ func dedupeWarnings(warnings []string) []string {
 	return uniqueWarnings
 }
 
+func getContextTfeRunInfo(ctx context.Context) (*TfeRunInfo, error) {
+	contextTfeRunInfo := ctx.Value(tfeRunInfoContextKey)
+	if contextTfeRunInfo != nil {
+		return contextTfeRunInfo.(*TfeRunInfo), nil
+	} else {
+		return nil, errors.New("Context does not have TFE run info")
+	}
+}
+
+func AddTfeRunInfoUrl(ctx context.Context, url string) {
+	info, err := getContextTfeRunInfo(ctx)
+	if err == nil {
+		info.AddTfeUrl(url)
+	}
```

Comment:
```suggestion
	if err =! nil {
	        log.debug(err) // or something like that
		return
	}
	info.AddTfeUrl(url)
```
for the same reasons as above, first deal with the error then proceed